base/shared/src/main/scala/scalaz/tc/semigroup.scala

```diff
@@ -1,11 +1,26 @@
 package scalaz
 package tc
 
+import scala.Int
 import java.lang.String
+
+import scala.annotation.tailrec
 import scala.language.experimental.macros
 
 trait SemigroupClass[A] {
   def mappend(a1: A, a2: => A): A
+
+  /**
+   * Compute the exponent of an element. a^n = exponent(a, n) = a * ... * a (n-times)
+   * If an algebra is idempotent, every element is an identity for itself modulo
+   * exponents, i.e. x*x = x and x^n = x - see [[Band]]
+   */
+  def exponent(a: A, i: Int): A = {
+    @tailrec
+    def go(a0: A, j: Int): A =
+      if (j == 0) a0 else go(mappend(a, a), j - 1)
+    go(a, i)
+  }
```

Comment:
The implementation doesn't look right.

Reply:
ah - it doesn't matter, but just for sanity's sake those should be `a0` in `mappend`